common/djangoapps/lang_pref/views.py

```diff
@@ -0,0 +1,23 @@
+"""
+Language Preference Views
+"""
+import json
+from django.conf import settings
+from django.views.decorators.csrf import csrf_exempt
+from django.utils.translation import LANGUAGE_SESSION_KEY
+from lang_pref import LANGUAGE_KEY
+from django.http import HttpResponse
+
+
+@csrf_exempt
+def update_language_session(request):
+    """
+    Update the language session key.
+    """
+    if request.method == 'PATCH':
+        # Setting the session language to the browser language, if it is supported.
```

Comment:
We don't really know where the language value comes from, so we should not assume it is coming from the browser.